gravitee-node-kubernetes/src/main/java/io/gravitee/node/kubernetes/keystoreloader/KubernetesConfigMapKeyStoreLoader.java

```diff
@@ -97,8 +92,41 @@ protected Completable init() {
         final List<Completable> locationObs = resources
             .keySet()
             .stream()
-            .map(location -> kubernetesClient.get(ResourceQuery.<ConfigMap>from(location).build()).flatMapCompletable(this::loadKeyStore))
-            .collect(Collectors.toList());
+            .map(location -> {
+                if (CERTIFICATE_FORMAT_PEM_REGISTRY.equals(location)) {
+                    String currentNamespace = KubernetesConfig.getInstance().getCurrentNamespace();
+                    return this.kubernetesClient.get(
+                            ResourceQuery
+                                .configMaps(currentNamespace)
+                                .labelSelector(
+                                    LabelSelector.equals(GRAVITEEIO_PEM_REGISTRY_LABEL, CERTIFICATE_FORMAT_PEM_REGISTRY.toLowerCase())
+                                )
+                                .build()
+                        )
+                        .map(ConfigMapList::getItems)
+                        .flatMapCompletable(configMaps ->
+                            Completable.merge(
```

Comment:
maybe from Cllable would be more suited for this case ?

Reply:
What do you mean exactly?